log4j-core/src/main/java/org/apache/logging/log4j/core/config/ConfigurationSource.java

```diff
@@ -71,16 +78,7 @@ public class ConfigurationSource {
      * @param file the file where the input stream originated
      */
     public ConfigurationSource(final InputStream stream, final File file) {
-        this.stream = Objects.requireNonNull(stream, "stream is null");
-        this.data = null;
-        this.source = new Source(file);
-        long modified = 0;
-        try {
-            modified = file.lastModified();
-        } catch (Exception ex) {
-            // There is a problem with the file. It will be handled somewhere else.
-        }
-        this.lastModified = modified;
+        this(null, new Source(file), stream, getLastModified(file.toPath()));
```

Comment:
If file is null this will result in an NPE in the previous version the NPE would be caught. That said, I don't recall if a null is ever passed here.